### PR TITLE
fix: Clear the child of the saved specs before clearing them

### PR DESCRIPTION
**Pull request by VladimirAmiorkov**
## PR Checklist

- [X] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [X] You have signed the [CLA](http://www.nativescript.org/cla). 
- [X] All existing tests are passing
- [ ] Tests for the changes are included

## What is the current behavior?
After adding a NativeScript `View` manually by calling `_addView()` in `RadListView`, calling `_removeView()` does not clear all of the references to that `View` and causes memory leak.

## What is the new behavior?
Calling `_removeView()` after manually adding a `View` via `_addView()` does not cause any `View` to be retained in the memory.

Fixes/Implements/Closes #[Issue Number].

Related to issue: https://github.com/telerik/nativescript-ui-feedback/issues/748

### CODE REVIEW

File: android/widgets/src/main/java/org/nativescript/widgets/GridLayout.java [hunk 1]
```diff
@@ -226,6 +226,7 @@ private void addToMap(View child) {
     }
 
     private void removeFromMap(View child) {
+        this.map.get(child).child = null;;
```
Comment (ADjenkov) [inline android/widgets/src/main/java/org/nativescript/widgets/GridLayout.java]: Multiple ';'

**Review by ADjenkov — CHANGES_REQUESTED** [review]

File: android/widgets/src/main/java/org/nativescript/widgets/GridLayout.java (hunk 1 above)
Comment (VladimirAmiorkov) [reply android/widgets/src/main/java/org/nativescript/widgets/GridLayout.java]: Ooo wow, I will fix it :)

**Comment by vakrilov** [comment]
Related to https://github.com/NativeScript/NativeScript/issues/6111

**Review by MartoYankov — APPROVED** [review]

**Comment by NathanaelA** [comment]
@vakrilov @MartoYankov @ADjenkov - Can we get this merged so that it can be tested out in a @next build; right now this is causing one of my clients a lot of pain from the memory leak it is causing in a RLV.

**Review by ADjenkov — APPROVED** [review]

**Comment by VladimirAmiorkov** [comment]
HI @NathanaelA ,

The PR is merged and is available in the **@next** version (4.3.0-2018-08-24-01)

**Comment by NathanaelA** [comment]
@VladimirAmiorkov - Thank you!

**Pull request merged**